### PR TITLE
adds Post routes

### PR DESCRIPTION
**Pull request by uimarshall**
For review

### CODE REVIEW

File: routes/api/posts.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
+const express = require('express');
+
+const router = express.Router();
+
+const secured = require('../../lib/middleware/secure');
+// const mongoose = require("mongoose");
```
Comment (enyeInc) [inline routes/api/posts.js]: deleted unused code

File: routes/api/posts.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    } catch (err) {
+        // Return empty list
+        return res.status(404).json({ nopost: 'No posts found!' });
+    }
+});
+
+// =================================================================================
+// GET a particular Post
+// Route: api/post/:id
+// Access: Not Protected route
```
Comment (enyeInc) [inline routes/api/posts.js:44]: i love this documentation, good job

File: routes/api/posts.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
+const express = require('express');
```
Comment (enyeInc) [inline routes/api/posts.js:1]: Go through these files and use (https://www.npmjs.com/package/http-status-codes) to replace your error code. then make sure that you are following the JSend Api when sending data back to the client (https://github.com/omniti-labs/jsend)

**Review by enyeInc — APPROVED** [review]
Code looks code, just make the convention changes and this is ready to be merged

**Review by olajohn-ajiboye — DISMISSED** [review]

**Review by enyeInc — CHANGES_REQUESTED** [review]
@uimarshall this PR is still waiting for your changes

File: routes/api/posts.js [hunk 4]
```diff
@@ -42,16 +65,16 @@ router.get('/:id', async (req, res) => {
...
     const { id } = req.user;
     try {
         const { errors, isValid } = validatePostInput(req.body);
 
         // Check Validation
         if (!isValid) {
-            // Return any errors with 400 status if not valid
-            return res.status(400).json(errors);
+            // Return any errors with 400(Bad Request) status if not valid
+            return res.status(BAD_REQUEST).json(errors);
```
Comment (Way2nnadi) [inline routes/api/posts.js]: `res.status(BAD_REQUEST).json({ errors });`

File: routes/api/posts.js [hunk 5]
```diff
@@ -103,15 +142,22 @@ router.post('/like/:id', secured(), async (req, res) => {
 
         // Check if user already like the post
         if (postFound.likes.filter(like => like.user.toString() === req.user.id).length > 0) {
-            return res.status(400).json({ alreadyliked: 'User already liked this post' });
+            return res.status(BAD_REQUEST).json({ alreadyliked: 'User already liked this post' });
```
Comment (Way2nnadi) [inline routes/api/posts.js]: `res.status(BAD_REQUEST).json({ message: 'User already liked this post' });`

File: routes/api/posts.js [hunk 6]
```diff
@@ -128,7 +174,7 @@ router.post('/unlike/:id', secured(), async (req, res) => {
 
         // Check if user have not yet like the post
         if (postFound.likes.filter(like => like.user.toString() === req.user.id).length === 0) {
-            return res.status(400).json({ notliked: 'You need to like this post first!' });
+            return res.status(BAD_REQUEST).json({ notliked: 'You need to like this post first!' });
```
Comment (Way2nnadi) [inline routes/api/posts.js]: `res.status(BAD_REQUEST).json({ message: 'You need to like this post first!' });`

File: routes/api/posts.js [hunk 7]
```diff
@@ -158,12 +210,11 @@ router.post('/comment/:id', secured(), async (req, res) => {
         // Check Validation
         if (!isValid) {
             // Return any errors with 400 status if not valid
-            return res.status(400).json(errors);
+            return res.status(BAD_REQUEST).json(errors);
```
Comment (Way2nnadi) [inline routes/api/posts.js]: `res.status(BAD_REQUEST).json({ errors });

File: routes/api/posts.js [hunk 8]
```diff
@@ -195,7 +253,7 @@ router.delete('/comment/:id/:comment_id', secured(), async (req, res) => {
             postFound.comments.filter(comment => comment.id.toString() === req.params.comment_id)
                 .length === 0
         ) {
-            return res.status(404).json({ commentnotexists: 'Comment does not exist' });
+            return res.status(BAD_REQUEST).json({ commentnotexists: 'Comment does not exist' });
```
Comment (Way2nnadi) [inline routes/api/posts.js]: `res.status(BAD_REQUEST).json({ message: 'Comment does not exist' });`

**Review by Way2nnadi — APPROVED** [review]

**Review by justiceotuya — APPROVED** [review]

**Pull request merged**